customcollection.go

```diff
@@ -0,0 +1,96 @@
+package goshopify
+
+import (
+	"fmt"
+)
+
+const customCollectionsBasePath = "admin/custom_collections"
+
+// CustomCollectionService is an interface for interacting with the custom
+// collection endpoints of the Shopify API.
+// See https://help.shopify.com/api/reference/customcollection
+type CustomCollectionService interface {
+	List(interface{}) ([]CustomCollection, error)
+	Count(interface{}) (int, error)
+	Get(int, interface{}) (*CustomCollection, error)
+	Create(CustomCollection) (*CustomCollection, error)
+	Update(int, CustomCollection) (*CustomCollection, error)
+	Delete(int) error
+}
+
+// CustomCollectionServiceOp handles communication with the custom collection
+// related methods of the Shopify API.
+type CustomCollectionServiceOp struct {
+	client *Client
+}
+
+// CustomCollection represents a Shopify custom collection.
+type CustomCollection struct {
+	ID             int    `json:"id"`
+	Handle         string `json:"handle"`
+	Title          string `json:"title"`
+	UpdatedAt      string `json:"updated_at"`
```

Comment:
You should be able to use a `*time.Time` type here. It works in most other cases, e.g. for [products](https://github.com/getconversio/go-shopify/blob/e461b55bc05f9b6e038e02d2b3fb91e1dc2d6d17/product.go#L37).